lib/omniauth/strategies/auth0.rb

```diff
@@ -81,6 +84,12 @@ def authorize_params
         %w[connection prompt].each do |key|
           params[key] = parsed_query[key] if parsed_query.key?(key)
         end
+
+        # Generate nonce if none exists
+        params['nonce'] = SecureRandom.hex if !params['nonce']
+        # Store authorize params in the session for token verification
+        session['authorize_params'] = params
```

Comment:
Is this SDK already using the session for storage? I don’t know about Ruby developer expectations, I just wonder if using the session limits developers ability to deploy in a distributed environment?